lib/src/over_react_redux/over_react_redux.dart

```diff
@@ -0,0 +1,393 @@
+    [
+      Function mapStateToProps,
+      dynamic mapDispatchToProps,
+      dynamic mergeProps,
+      JsConnectOptions options,
+    ]
+  );
+
+@visibleForTesting
+ReactClass Function(ReactClass) Function(
+    [
+      Function mapStateToProps,
+      dynamic mapDispatchToProps,
+      dynamic mergeProps,
+      JsConnectOptions options,
+    ]
+  ) mockableJsConnect = _jsConnect;
+
+@JS('ReactRedux')
+class JsReactRedux {
+  external static ReactClass get Provider;
+  external static ReactContext get ReactReduxContext;
+}
+
+class ReduxProviderProps extends component_base.UiProps
+    with
+        builder_helpers.GeneratedClass
+    implements
+        builder_helpers.UiProps {
+  // Initialize to a JsBackedMap so that copying can be optimized
+  // when converting props during ReactElement creation.
+  // TODO 3.0.0-wip generate JsBackedMap-based implementation used when no backing map is provided, like we do for Component2
+  ReduxProviderProps([Map props]) : this.props = props ?? new JsBackedMap();
+
+  @override
+  final Map props;
+
+  @override
+  ReactComponentFactoryProxy get componentFactory => ReactJsReactReduxComponentFactoryProxy(JsReactRedux.Provider, isProvider: true);
+
+  @override
+  String get propKeyNamespace => '';
+
+  Store get store => props['store'];
+  set store(Store v) => props['store'] = v;
+
+  dynamic get context => props['context'];
+  set context(dynamic v) => props['context'] = v;
+}
+
+UiFactory<ReduxProviderProps> ReduxProvider = ([Map map]) => new ReduxProviderProps(map);
+
+class ReactJsReactReduxComponentFactoryProxy extends ReactJsContextComponentFactoryProxy {
+  /// The JS class used by this factory.
+  @override
+  final ReactClass type;
+
+  @override
+  final Function factory;
+
+  @override
+  final bool shouldConvertDomProps;
+
+  @override
+  final bool isConsumer;
+
+  @override
+  final bool isProvider;
+
+  ReactJsReactReduxComponentFactoryProxy(
+    ReactClass jsClass, {
+    this.shouldConvertDomProps: true,
+    this.isConsumer: false,
+    this.isProvider: false,
+  }) : this.type = jsClass,
+        this.factory = React.createFactory(jsClass),
+        super(jsClass, isProvider: isProvider, isConsumer: isConsumer, shouldConvertDomProps: shouldConvertDomProps);
+
+  @override
+  ReactElement build(Map props, [List childrenArgs]) {
+    return super.build(_generateReduxJsProps(props), childrenArgs);
+  }
+
+  /// Returns a JavaScript version of the specified [props], preprocessed for consumption by ReactJS and prepared for
+  /// consumption by the [react] library internals.
+  Map _generateReduxJsProps(Map props) {
+    JsBackedMap propsForJs = new JsBackedMap.from(props);
+
+    if (propsForJs['store'] != null) {
+      propsForJs['store'] = _reduxifyStore(propsForJs['store']);
+    }
+
+    if (propsForJs['context'] != null) {
+      propsForJs['context'] = propsForJs['context'].jsThis;
+    }
+
+    return propsForJs;
+  }
+}
+
+_reduxifyStore(Store store){
```

Comment:
This could use a doc comment and return type annotation
```suggestion
JsReactReduxStore _reduxifyStore(Store store) {
```